### PR TITLE
Initial experimental implementation of RIVA 128 emulation.

### PR DESCRIPTION
**Pull request by MoochMcGee**
VESA, PCI configuration, PTIMER, and interrupts all work.

### CODE REVIEW

**Comment by MoochMcGee** [comment]
@dhrdlicka It's fixed, sorry, it was my editor assuming CRLF newlines instead of the standard LF.

**Comment by MoochMcGee** [comment]
Sorry, I didn't mean to request another review lol

**Comment by TC1995** [comment]
How about the accel? That would be the real progress.

**Comment by MoochMcGee** [comment]
@TC1995 nVidia hardware acceleration is extremely complicated, so it'll take a lot more time than this has. Please be patient.

**Comment by darkstar** [comment]
> @dhrdlicka It's fixed, sorry, it was my editor assuming CRLF newlines instead of the standard LF.

try using `core.autocrlf=input` if you're on windows. That's what I use

File: src/video/vid_riva128.c [hunk 1]
```diff
@@ -0,0 +1,1072 @@
...
+        mem_mapping_disable(&riva128->mmio_mapping);
+
+    pclog("riva128->lfb_base = %08X\n", riva128->lfb_base);
+    if (riva128->lfb_base) {
+	mem_mapping_set_addr(&riva128->linear_mapping, riva128->lfb_base, 0x0c00000);
+	mem_mapping_set_addr(&riva128->ramin_mapping, riva128->lfb_base + 0x0c00000, 0x0400000);
+    } else {
+        mem_mapping_disable(&riva128->linear_mapping);
+        mem_mapping_disable(&riva128->ramin_mapping);
+    }
```
Comment (darkstar) [inline src/video/vid_riva128.c]: check indenting

Comment (MoochMcGee) [reply src/video/vid_riva128.c]: The indenting looks fine to me. :/

File: src/video/vid_riva128.c [hunk 2]
```diff
@@ -0,0 +1,1072 @@
...
+
+    if ((addr >= 0x1800) && (addr <= 0x18ff))
+	return riva128_pci_read(0,addr & 0xff,p);
+
+    switch(addr) {
+	case 0x6013b4: case 0x6013b5:
+	case 0x6013d4: case 0x6013d5:
+	case 0x6013da:
+	case 0x0c03c2: case 0x0c03c3: case 0x0c03c4: case 0x0c03c5:
+	case 0x6813c6: case 0x6813c7: case 0x6813c8: case 0x6813c9: case 0x6813ca: case 0x6813cb: case 0x6813cc:
```
Comment (darkstar) [inline src/video/vid_riva128.c]: IMHO multiple lines for the case labels would be clearer but that's just my opinion

File: src/video/vid_riva128.c [hunk 3]
```diff
@@ -0,0 +1,1072 @@
...
+
+void
+riva128_mmio_write(uint32_t addr, uint8_t val, void *p)
+{
+    uint32_t tmp;
+
+    addr &= 0xffffff;
+
+    if (addr != 0x6013d4 && addr != 0x6013d5 && addr != 0x6013b4 && addr != 0x6013b5 && addr != 0x6013da && !((addr >= 0x6813c6) && (addr <= 0x6813cc))) {
+	tmp = riva128_mmio_read_l(addr,p);
```
[the remark this answers is not in the document]
Comment (MoochMcGee) [reply src/video/vid_riva128.c]: The magic addresses are derived from the VGA register addresses. I will split the if over 2 lines though, thanks.

File: src/video/vid_riva128.c [hunk 4]
```diff
@@ -0,0 +1,1072 @@
...
+                        {
+                                .description = ""
+                        }
+                },
+                .default_int = 4
+        },
+        {
+                .type = -1
+        }
+};
```
Comment (darkstar) [inline src/video/vid_riva128.c]: yet another indenting style. try sticking with one style. There are linters out there that can do that automatically for you

File: src/video/vid_riva128.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (darkstar) [reply src/video/vid_riva128.c]: you could use a macro and have that build the constants from its components. Like `MMIO_ADDR(foo, bar, baz)` or something. That would show where they come from

File: src/video/vid_riva128.c (hunk 1 above)
Comment (darkstar) [reply src/video/vid_riva128.c]: hm. for me it looks like
```
if (riva128->lfb_base) {
mem_mapping_set_addr(&riva128->linear_mapping, riva128->lfb_base, 0x0c00000);
} else {
  mem_mapping_disable(&riva128->linear_mapping);
}
```
i.e. the "else" is indented by 4 spaces, the "if" part not at all. (or rather 4 spaces vs. 8 spaces, if you look at the total indentation)

**Comment by Syon1** [comment]
Any progress on this?

**Comment by MoochMcGee** [comment]
Moved this away from my master branch so I gotta close this.

**Pull request closed without merging**